staging/src/k8s.io/dynamic-resource-allocation/structured/internal/experimental/allocator_experimental.go

```diff
@@ -1202,6 +1180,11 @@ func (alloc *allocator) selectorsMatch(r requestIndices, device *draapi.Device,
 //
 // The rollback function is only provided in case of a successful allocation
 // (true and no error).
+//
+// r identifies the request for which allocation is attempted.
+// The device is the candidate under consideration.
+// If "must" is true, then it is okay (but not required) to return an error which describes in more detail why allocation wasn't possible.
+// This is used when the caller wants exactly this device and will give up if it cannot allocate it.
```

Comment:
I think a better place is after the initial paragraph (parameters first, then return values).